making TORCH_CHECK_{COND} non-fatal

dolpm opened the pull request:
TORCH_CHECK is non-fatal by design, but TORCH_CHECK_{COND} macros are fatal. this is confusing, and we should limit fatality to the set of debug macros.

Differential Revision: D86168955

malfet reviewed (approved):
LGTM, provided CI is green (please check bazel failures, Windows one feels false positive)

dolpm committed:
make TORCH_CHECK_{COND} non-fatal (#167004)

Summary:

TORCH_CHECK is non-fatal by design, but TORCH_CHECK_{COND} macros are fatal. this is confusing, and we should limit fatality to the set of debug macros.

Test Plan: ci

Reviewed By: malfet, ivy-zhou

Differential Revision: D86168955
